ycmd/tests/get_completions_test.py

```diff
@@ -115,9 +123,800 @@ def GetCompletions_IdentifierCompleter_WorksForSpecialIdentifierChars_test():
                            completion_data ).json[ 'completions' ]
 
   assert_that( results,
-               has_items( CompletionEntryMatcher( 'font-size' ),
-                          CompletionEntryMatcher( 'font-family' ) ) )
+               has_items( CompletionEntryMatcher( 'font-size', '[ID]' ),
+                          CompletionEntryMatcher( 'font-family', '[ID]' ) ) )
+
 
+@with_setup( Setup )
+def GetCompletions_Fall_Back_ClangCompleter_test():
+
+  # because we use it a lot
+  NO_COMPLETIONS_ERROR = ErrorMatcher( RuntimeError, NO_COMPLETIONS_MESSAGE )
+
+  """
+  GetCompletions_Fall_Back_test
+
+  Tests for:
+    - [X] multiple languages: (use cases)
+      - [X] C:      macros
+      - [X] C++:    templates
+      - [X] Python: properties set from external method
+    - [X] Forced vs. Triggered semantic completion
+    - [X] Identifiers tagged with '[ID]'
+    - [X] Filter/sort based on query
+    - [X] Min chars for completion (only applies to ID)
+    - [X] Extra conf throws exception
```

Comment:
I’m going to trim down the tests to just:

- one language (clang completer-based)
- forced vs triggered semantic (fallback only in 2nd case)
- exception thrown, vs. just no completions for query
- min chars for identifier completion

Would anyone object to keeping the testdata files as they are? This is not just laziness, but I do find it useful to open the testsdata/* files to do manual testing of a particular feature (almost as a demo). And the test files i wrote mostly for manual testing in the first instance to ensure the feature feels right and works sensibly.